js/src/components/RelatedKeyphrasesModalContent.js

```diff
@@ -10,123 +11,132 @@ import KeyphrasesTable from "./modals/KeyphrasesTable";
 import SemRushUpsellAlert from "./modals/SemRushUpsellAlert";
 import SemRushRequestFailed from "./modals/SemRushRequestFailed";
 
+/**
+ * Determines whether the error property is present in the passed response object.
+ *
+ * @param {Object} response The response object.
+ *
+ * @returns {boolean} Whether or not the error property is present.
+ */
+function hasError( response ) {
+	return response && "error" in response;
+}
+
+/**
+ * Gets a user message based on the passed props' values.
+ *
+ * @param {Object} props The props to use.
+ *
+ * @returns {wp.Element} The user message.
+ */
+function getUserMessage( props ) {
+	const {
+		isPending,
+		requestLimitReached,
+		isSuccess,
+		response,
+		requestHasData,
+	} = props;
+
+	if ( isPending ) {
+		return <SemRushLoading />;
+	}
+
+	if ( requestLimitReached ) {
+		return <SemRushLimitReached />;
+	}
+
+	if ( ! isSuccess && hasError( response ) ) {
+		return <SemRushRequestFailed />;
+	}
+
+	if ( ! requestHasData ) {
+	   return <p> { __( "Sorry, there's no data available for that keyphrase/country combination.", "wordpress-seo" ) } </p>;
+	}
+}
+
+/**
+ * Whether the request limit is reached or the request has failed.
```

Comment:
I think we're aiming for descriptions that start with a vert indicating what the function does, e.g. "Determines..."